components/infra-proxy-service/server/nodes.go

```diff
@@ -53,6 +57,64 @@ func (s *Server) GetAffectedNodes(ctx context.Context, req *request.AffectedNode
 	}, nil
 }
 
+// GetNode fetches the node from chef infra server
+func (s *Server) GetNode(ctx context.Context, req *request.Node) (*response.Node, error) {
+	err := validation.New(validation.Options{
+		Target:          "node",
+		Request:         *req,
+		RequiredDefault: true,
+	}).Validate()
+
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.client.Nodes.Get(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	defaultAttributes, err := json.Marshal(res.DefaultAttributes)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	automaticAttributes, err := json.Marshal(res.AutomaticAttributes)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	normalAttributes, err := json.Marshal(res.NormalAttributes)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	overrideAttributes, err := json.Marshal(res.OverrideAttributes)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	// TODO: Chef Infra Server node detail API is not returning the node ID https://docs.chef.io/api_chef_server/#get-38
+	return &response.Node{
+		NodeId:              res.Name,
```

Comment:
Chef Infra Server Node detail API does not return the node ID, so consider node name as node ID here.